mcstatus/motd.py

```diff
@@ -0,0 +1,753 @@
+from __future__ import annotations
+
+import abc
+import re
+import typing as t
+from collections.abc import Callable, Iterable
+from dataclasses import dataclass
+from enum import Enum
+
+if t.TYPE_CHECKING:
+    from typing_extensions import Self, TypeAlias
+
+    from mcstatus.pinger import RawResponseDescription, RawResponseDescriptionWhenDict
+else:
+    RawResponseDescription = dict
+    RawResponseDescriptionWhenDict = dict
+
+__all__ = [
```

Comment:
If we assume that user can heavily modify behavior, maybe better would be to not do any private items in this module?

Reply:
I would vote for having items not be private as well

Reply:
Well, these are different things. Currently, such classes as transformations have only one public method, and a couple of privates. These privates can be modified by a user, but should not be called directly. So in result, this is still a part of public API, but they're private and limited to only this class.

It would be better, if we had docs, and I could reference there what is public, and what is private.

Reply:
With the current addition of transformers etc. I don't think this is relevant anymore, is it?